wandb/sdk/internal/datastore.py

```diff
@@ -274,7 +294,7 @@ def write(self, obj):
         ret = self._write_data(s)
```

Comment:
maybe add an assertion about the offsets?

Reply:
what type of assertion were you thinking about.. i think most should be covered in the _write_data area

Reply:
just ordering... basically that start_offset is <= end_offset and so on

Reply:
i think the write path is good.. but i can imagine some more read assertions might be useful. ill think about that and add with a follow on testing PR if it makes it.